### PR TITLE
Added parameter description  to all functions and documentation to some in Functional.hs for cleanup

### PR DESCRIPTION
**Pull request by championballer**

### CODE REVIEW

File: hasktorch/src/Torch/Functional.hs [hunk 1]
```diff
@@ -478,7 +478,11 @@ ne a b = unsafePerformIO $ (cast2 ATen.ne_tt) a b
 (/=.) = ne
 
 -- | toDType
-toDType :: DType -> Tensor -> Tensor
+-- Casting to given Dtype, where Dtype is an object that represents the data type of a tensor in hasktorch
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
-- Casting to given 'DType', where 'DType' represents the data type of a tensor in hasktorch.
```

File: hasktorch/src/Torch/Functional.hs [hunk 2]
```diff
@@ -478,7 +478,11 @@ ne a b = unsafePerformIO $ (cast2 ATen.ne_tt) a b
...
+toDType 
+ :: DType -- ^ datatype to cast to
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
 :: DType -- ^ data type to cast to
```

File: hasktorch/src/Torch/Functional.hs [hunk 3]
```diff
@@ -505,12 +509,20 @@ binaryCrossEntropyLoss'
  -> Tensor -- ^ output
 binaryCrossEntropyLoss' t target = unsafePerformIO $ (cast4 ATen.binary_cross_entropy_tttl) t target (onesLike target) ReduceMean
 
--- | mse_loss
-mseLoss :: Tensor -> Tensor -> Tensor
+-- | mseLoss
+-- Creates a criterion that measures the mean squared error (squared L2 norm) between each element in the input xx and target yy .
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
-- Creates a criterion that measures the mean squared error (squared L2 norm) between each element in the @input@ and @target@.
```

File: hasktorch/src/Torch/Functional.hs [hunk 4]
```diff
@@ -505,12 +509,20 @@ binaryCrossEntropyLoss'
...
+mseLoss 
+ :: Tensor -- ^ input
+ -> Tensor -- ^ target
+ -> Tensor -- ^ output
 mseLoss a b = unsafePerformIO $ (cast3 ATen.mse_loss_ttl) a b ATen.kMean
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
mseLoss input target = unsafePerformIO $ (cast3 ATen.mse_loss_ttl) input target ATen.kMean
```

File: hasktorch/src/Torch/Functional.hs [hunk 5]
```diff
@@ -591,26 +603,57 @@ maxPool3d kernelSize stride padding dilation ceilMode self =
     unsafePerformIO $ (cast6 ATen.max_pool3d_tllllb)
         self kernelSize stride padding dilation ceilMode
 
+-- | inverse
+-- Takes the inverse of the square matrix input. input can be batches of 2D square tensors, in which case this function would return a tensor composed of individual inverses.
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
-- Takes the inverse of the square matrix input. @input@ can be batches of 2D square tensors, in which case this function would return a tensor composed of individual inverses.
```

File: hasktorch/src/Torch/Functional.hs [hunk 6]
```diff
@@ -839,19 +891,31 @@ geqrf
 geqrf t = unsafePerformIO $ (cast1 ATen.geqrf_t) t
 
 
--- | Computes the orthogonal matrix Q of a QR factorization, from the (input, input2) tuple returned by torch.geqrf().
+-- | Computes the orthogonal matrix Q of a QR factorization, from the (input, input2) tuple returned by geqrf function.
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
-- | Computes the orthogonal matrix Q of a QR factorization, from the @(input, input2)@ tuple returned by 'geqrf' function.
```

File: hasktorch/src/Torch/Functional.hs [hunk 7]
```diff
@@ -839,19 +891,31 @@ geqrf
...
 -- This directly calls the underlying LAPACK function ?orgqr. See LAPACK documentation for orgqr for further details.
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
-- This directly calls the underlying LAPACK function @?orgqr@. See LAPACK documentation for @orgqr@ for further details.
```

File: hasktorch/src/Torch/Functional.hs [hunk 8]
```diff
@@ -839,19 +891,31 @@ geqrf
...
-orgqr :: Tensor -> Tensor -> Tensor
+orgqr 
+ :: Tensor -- ^ the a from geqrf function
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
 :: Tensor -- ^ the @a@ from @geqrf@ function
```

File: hasktorch/src/Torch/Functional.hs [hunk 9]
```diff
@@ -839,19 +891,31 @@ geqrf
...
+ -> Tensor -- ^ the tau from geqrf function
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
 -> Tensor -- ^ the @tau@ from @geqrf@ function
```

File: hasktorch/src/Torch/Functional.hs [hunk 10]
```diff
@@ -839,19 +891,31 @@ geqrf
...
+ -> Tensor -- ^ output
 orgqr b a = unsafePerformIO $ (cast2 ATen.orgqr_tt) b a
 
+-- | sign
+-- Returns a new tensor with the signs of the elements of input
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
-- Returns a new tensor with the signs of the elements of @input@
```

File: hasktorch/src/Torch/Functional.hs [hunk 11]
```diff
@@ -839,19 +891,31 @@ geqrf
...
 sign 
     :: Tensor -- ^ input
     -> Tensor -- ^ output
 sign t = unsafePerformIO $ (cast1 ATen.sign_t) t
 
-transpose :: Tensor -> Int -> Int -> Tensor
+-- | transpose
+-- Returns a tensor that is a transposed version of input. The given dimensions dim0 and dim1 are swapped.
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]:
```suggestion
-- Returns a tensor that is a transposed version of @input@. The given dimensions @dim0@ and @dim1@ are swapped.
```

**Comment by championballer** [comment]
@tscholak Gotcha boss. Thanks for the review. On it. :)

**Comment by championballer** [comment]
@tscholak @austinvhuang Made the changes requested along with removal for redundant function name annotation. Please review for any modifications. Thanks. :)

File: hasktorch/src/Torch/Functional.hs [hunk 12]
```diff
@@ -251,89 +238,77 @@ matmul
...
--- Applies element-wise, \text{SELU}(x) = scale * (\max(0,x) + \min(0, \alpha * (\exp(x) - 1)))SELU(x)=scale∗(max(0,x)+min(0,α∗(exp(x)−1))) , with \alpha=1.6732632423543772848170429916717α=1.6732632423543772848170429916717 and scale=1.0507009873554804934193349852946scale=1.0507009873554804934193349852946 .
+-- | Applies element-wise, \text{SELU}(x) = scale * (\max(0,x) + \min(0, \alpha * (\exp(x) - 1)))SELU(x)=scale∗(max(0,x)+min(0,α∗(exp(x)−1))) , with \alpha=1.6732632423543772848170429916717α=1.6732632423543772848170429916717 and scale=1.0507009873554804934193349852946scale=1.0507009873554804934193349852946 .
 selu 
     :: Tensor -- ^ input
     -> Tensor -- ^ output
 selu t = unsafePerformIO $ (cast1 ATen.selu_t) t
 
--- | celu
--- Applies element-wise, \text{CELU}(x) = \max(0,x) + \min(0, \alpha * (\exp(x/\alpha) - 1))CELU(x)=max(0,x)+min(0,α∗(exp(x/α)−1)) .
+-- | Applies element-wise, \text{CELU}(x) = \max(0,x) + \min(0, \alpha * (\exp(x/\alpha) - 1))CELU(x)=max(0,x)+min(0,α∗(exp(x/α)−1)) .
```
Comment (tscholak) [inline hasktorch/src/Torch/Functional.hs]: not sure if latex renders correctly. it's ok now, but be prepared to revise once this is published at hasktorch.github.io

Comment (championballer) [reply hasktorch/src/Torch/Functional.hs]: Sure.

**Pull request merged**